src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/DispatchOperationRuntimeHelpers.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Linq;
+
+namespace CoreWCF.Dispatcher;
+
+public static class DispatchOperationRuntimeHelpers
+{
+    internal static Dictionary<string, IOperationInvoker> OperationInvokers { get; } = new();
+
+    public static void RegisterOperationInvoker(string key, IOperationInvoker invoker)
+    {
+        OperationInvokers[key] = invoker;
+    }
+
+    internal static string GetKey(MethodInfo method)
+    {
+        StringBuilder stringBuilder = new StringBuilder($"{method.DeclaringType.FullName}.{method.Name}(");
+        stringBuilder.Append(string.Join(", ", method.GetParameters().Select(GetParameterString)));
+        foreach (var keyValuePair in s_IntegratedTypesMap)
+        {
+            stringBuilder.Replace(keyValuePair.Key, keyValuePair.Value);
+        }
```

Comment:
This is potentially very expensive. It would be cheaper to do this in GetParameterFullName. Then you could lookup the type in the dictionary instead of the type name. If this is going to be aot only, then you can use Dictionary<type,string>, but if it's going to be used in regular jit scenarios too, Type might not be stable when using hot reload. I know we had a problem with MethodInfo in the WCF client when using hot reload. In that case, use Type.TypeHandle.Value to get the type handle value and use Dictionary<IntPtr, string>

Reply:
done here https://github.com/CoreWCF/CoreWCF/pull/1227/commits/75c082b1d373ce0322eb9eff810f58aa45ef1180